packages/gatsby-source-contentful/src/fetch.js

```diff
@@ -19,7 +19,8 @@ module.exports = async ({ spaceId, accessToken, host, syncToken, cacheDir, envir
   console.time(`Fetch Contentful data`)
 
   // If the user knows they are offline, serve them cached result
-  if (process.env.GATSBY_CONTENTFUL_OFFLINE === `true`) {
+  // For prod builds though always fail if we can't get the latest data
+  if (process.env.GATSBY_CONTENTFUL_OFFLINE === `true` && process.env.ACTIVE_ENV === `development`) {
```

Comment:
maybe
```diff
-  if (process.env.GATSBY_CONTENTFUL_OFFLINE === `true` && process.env.ACTIVE_ENV === `development`) {
+  if (process.env.GATSBY_CONTENTFUL_OFFLINE === `true` && process.env.NODE_ENV !== `production`) {
```
?
But yeah, we should probably make it easier to determine if we are in `build` or `develop`